tests/foreman/destructive/test_packages.py

```diff
@@ -42,19 +44,49 @@ def test_positive_all_packages_update(target_sat):
     :customerscenario: true
     """
     # Register to CDN for package updates
-    target_sat.register_to_cdn()
+    sat_maintain.register_to_cdn()
     # Update packages with yum
-    result = target_sat.execute('yum update -y --disableplugin=foreman-protector')
+    result = sat_maintain.execute('yum update -y --disableplugin=foreman-protector')
     assert result.status == 0
     # Reboot
-    if target_sat.execute('needs-restarting -r').status == 1:
-        target_sat.power_control(state='reboot')
+    if sat_maintain.execute('needs-restarting -r').status == 1:
+        sat_maintain.power_control(state='reboot')
     # Run check-update again to verify there are no more packages available to update
-    result = target_sat.cli.Packages.check_update()
+    result = sat_maintain.cli.Packages.check_update()
     # Regex to match if there are packages available to update
     # Matches lines like '\n\nwalrus.noarch        5.21-1        custom_repo\n'
     pattern = '(\\n){1,2}(\\S+)(\\s+)(\\S+)(\\s+)(\\S+)(\\n)'
     matches = re.search(pattern, result.stdout)
     assert matches is None  # No packages available to update
     assert 'FAIL' not in result.stdout
     assert result.status == 0
+
+
+@pytest.mark.include_capsule
+def test_negative_remove_satellite_packages(sat_maintain):
+    """Ensure user can't remove satellite or its dependent packages
+
+    :id: af150302-418a-4d42-8d01-bb0e6b90f81f
+
+    :steps:
+        1. yum remove <satellite or other dependency>
+
+    :expectedresults: removal should fail due to protecting the satellite package
+
+    :BZ: 1884395
+
+    :customerscenario: true
+    """
+    # Packages include satellite direct dependencies like foreman,
+    # but also dependency of dependencies like wget for foreman
+    if isinstance(sat_maintain, Satellite):
+        package_list = ['foreman', 'foreman-proxy', 'katello', 'wget', 'satellite']
+    else:
+        package_list = ['foreman-proxy', 'satellite-capsule']
```

Comment:
What would you say to use `product_rpm_name` and `upstream_rpm_name` instead?

https://github.com/SatelliteQE/robottelo/blob/783549a099f80c1c3c4296058ad09853675ec3c7/robottelo/hosts.py#L1486-L1489

https://github.com/SatelliteQE/robottelo/blob/783549a099f80c1c3c4296058ad09853675ec3c7/robottelo/hosts.py#L1717-L1719

Reply:
That still doesn't account for things like foreman isn't on a capsule

Reply:
hmm, that is probably not worth the effort. My main point was that the names of some of the packages that you use in the test are already defined in the Host classes. Generally, I strive for de-duplication of constant-like strings as much as possible.

Reply:
Oh ok so you were saying to do this?
```suggestion
    if isinstance(sat_maintain, Satellite):
        package_list = ['foreman', 'foreman-proxy', 'katello', 'wget', sat_maintain.product_rpm_name]
    else:
        package_list = ['foreman-proxy', sat_maintain.product_rpm_name]
```

Reply:
Yes, sort of, but the more I look into it, it seems that my suggestion does not bring too much value to anything. Especially when RPM names of the core product packages do not change. We should be fine with your implementation.